packages/fiori-mcp-server/test/integration/assertions/snapshots.ts

```diff
@@ -0,0 +1,300 @@
+        if (
+            current === null ||
+            typeof current !== 'object' ||
+            !(key in (current as Record<string | number, unknown>))
+        ) {
+            return undefined;
+        }
+
+        current = (current as Record<string | number, unknown>)[key];
+    }
+
+    return current;
+}
+
+/**
+ * Compare two json objects or string.
+ *
+ * @param obj1 First json object or string to compare.
+ * @param obj2 Second json object or string to compare.
+ * @returns A string describing the differences, or `undefined` if the strings are identical.
+ */
+function compareJson(obj1: string | object, obj2: string | object): string | undefined {
+    const diffChanges = diffJson(obj1, obj2);
+    const diffResultString = getDiffResultString(diffChanges);
+    return diffResultString ? diffResultString : undefined;
+}
+
+/**
+ * Recursively checks whether all properties and values of the `expected` object
+ * are contained within the `actual` object.
+ *
+ * @param expected The reference value or object to compare against.
+ * @param actual The object or value being tested for containment.
+ * @param path (Internal) The current object path used for detailed mismatch reporting.
+ * @returns A descriptive error string indicating the first mismatch or missing key,
+ *          or `undefined` if `actual` fully contains `expected`.
+ */
+function deepContains(expected: unknown, actual: unknown, path = ''): string | undefined {
+    // Handle primitive values (and null)
+    if (typeof expected !== 'object' || expected === null || typeof actual !== 'object' || actual === null) {
+        return Object.is(expected, actual) ? undefined : `Mismatch at ${path}: expected ${expected}, got ${actual}`;
+    }
+
+    // At this point both are non-null objects
+    const expectedObj = expected as Record<string, unknown>;
+    const actualObj = actual as Record<string, unknown>;
+
+    for (const key of Object.keys(expectedObj)) {
+        if (!(key in actualObj)) {
+            return `Missing key at ${path}.${key}`;
+        }
+
+        const result = deepContains(expectedObj[key], actualObj[key], `${path}.${key}`);
+        if (result) {
+            return result;
+        }
+    }
+
+    // Ignore extra keys in `actual`
+    return undefined;
+}
+
+/**
+ * Compare two strings.
+ *
+ * @param obj1 First object to compare.
+ * @param obj2 Second object to compare.
+ * @returns A string describing the differences, or `undefined` if the strings are identical.
+ */
+function compareStrings(obj1: string, obj2: string): string | undefined {
+    const diffChanges = diffTrimmedLines(obj1, obj2);
+    const diffResultString = getDiffResultString(diffChanges);
+    return diffResultString ? diffResultString : undefined;
+}
+
+/**
+ * Get the diff results as colored string.
+ *
+ * @param diffChanges Array of changes, result from diff.
+ * @returns Diff results as colored string.
+ */
+function getDiffResultString(diffChanges: Change[]): string {
```

Comment:
these seems asserting testing api. can we re-use from jest api?

same for `compareStrings`

Reply:
I did not find clean way - here we do not use jest to run tests. Also I checked promptfoo, but did not find anything building to use for object/file content comparation with diff output

Reply:
Maybe this one https://nodejs.org/api/assert.html help?